shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -450,28 +482,27 @@ static bool ClipBoundsContainsPlatformViewBoundingRect(const SkRect& clip_bounds
         break;
       }
       case kClipRect: {
-        if (ClipBoundsContainsPlatformViewBoundingRect((*iter)->GetRect(), bounding_rect,
-                                                       transformMatrix)) {
+        if (ClipRectContainsPlatformViewBoundingRect((*iter)->GetRect(), bounding_rect,
+                                                     transformMatrix)) {
           break;
         }
         [maskView clipRect:(*iter)->GetRect() matrix:transformMatrix];
         clipView.maskView = maskView;
         break;
       }
       case kClipRRect: {
-        if (ClipBoundsContainsPlatformViewBoundingRect((*iter)->GetRRect().getBounds(),
-                                                       bounding_rect, transformMatrix)) {
+        if (ClipRRectContainsPlatformViewBoundingRect((*iter)->GetRRect(), bounding_rect,
+                                                      transformMatrix)) {
           break;
         }
         [maskView clipRRect:(*iter)->GetRRect() matrix:transformMatrix];
         clipView.maskView = maskView;
         break;
       }
       case kClipPath: {
-        if (ClipBoundsContainsPlatformViewBoundingRect((*iter)->GetPath().getBounds(),
-                                                       bounding_rect, transformMatrix)) {
-          break;
-        }
+        // TODO(cyanglaz): Find a way to pre-determine if path contains the PlatformView boudning
```

Comment:
this is probably a hard geometry problem. but i bet most clipping actions are just simple shapes like RRect